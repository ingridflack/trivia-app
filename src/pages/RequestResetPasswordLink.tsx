import { Box, Card, Container, Image, Text } from "@chakra-ui/react";
import SendEmailForm from "../components/SendEmailForm";
import Header from "../components/Header";

export default function RequestResetPasswordLink() {
  return (
    <>
      <Header />

      <Container
        display="flex"
        alignItems="center"
        justifyContent="center"
        minH="calc(var(--chakra-vh) - 50px)" // 100vh - 50px (height of the header)
      >
        <Card
          display="flex"
          flexDirection={["column", "column", "row"]}
          justifyContent="space-around"
          width="100%"
          padding="70px"
          shadow="xl"
        >
          <Box maxW="392px" width="100%">
            <Text fontSize="xx-large" marginBottom="30px" color="gray.700">
              Password recovery
            </Text>
            <SendEmailForm />
          </Box>

          <Image
            src="/src/assets/recovery.svg"
            alt="Login image"
            width="325px"
            height="324px"
            pointerEvents="none"
          />
        </Card>
      </Container>
    </>
  );
}
