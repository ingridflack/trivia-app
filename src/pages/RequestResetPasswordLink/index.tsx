import { Box, Card, Container, Image, Text, useToast } from "@chakra-ui/react";
import SendEmailForm from "../../components/SendEmailForm/SendEmailForm";
import Header from "../../components/Header";
import { useState } from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import * as AuthService from "../../services/authService";
import { Footer } from "../../components/Footer";

export interface SendEmailFormValues {
  email: string;
}

export default function RequestResetPasswordLink() {
  const {
    handleSubmit,
    register,
    formState: { errors, isSubmitting },
  } = useForm<SendEmailFormValues>();
  const toast = useToast();
  const [successfullySent, setSuccessfullySent] = useState(false);

  const onSubmit: SubmitHandler<SendEmailFormValues> = async (values) => {
    try {
      await AuthService.requestResetPasswordLink(values.email);

      toast({
        title: "Email sent.",
        description: "Check your inbox.",
        status: "success",
        duration: 2000,
        isClosable: true,
        position: "top-right",
      });

      setSuccessfullySent(true);
    } catch (error) {
      console.log(error);
      toast({
        title: "An error occurred.",
        description: "Unable to login.",
        status: "error",
        duration: 2000,
        isClosable: true,
        position: "top-right",
      });
    }
  };
  return (
    <>
      <Header />

      <Container
        as="main"
        display="flex"
        alignItems="center"
        justifyContent="center"
        paddingY="32px"
      >
        <Card
          display="flex"
          flexDirection={["column", "column", "row"]}
          justifyContent="space-around"
          width="100%"
          padding={{ base: "20px", md: "32px" }}
          shadow="xl"
          maxW="635px"
        >
          <Box maxW="392px" width="100%">
            <Text fontSize="xx-large" marginBottom="32px" color="gray.700">
              Password recovery
            </Text>
            <SendEmailForm
              onSubmit={handleSubmit(onSubmit)}
              successfullySent={successfullySent}
              errors={errors}
              isSubmitting={isSubmitting}
              register={register}
            />
          </Box>

          <Image
            src="/assets/recovery.svg"
            alt="Login image"
            width="325px"
            height="324px"
            pointerEvents="none"
            display={{ base: "none", md: "none", lg: "block" }}
          />
        </Card>
      </Container>
      <Footer />
    </>
  );
}
