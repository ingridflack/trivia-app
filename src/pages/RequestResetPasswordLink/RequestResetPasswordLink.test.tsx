import { describe, expect, it } from "vitest";
import RequestResetPasswordLink from ".";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("RequestResetPasswordLink", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <RequestResetPasswordLink />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
