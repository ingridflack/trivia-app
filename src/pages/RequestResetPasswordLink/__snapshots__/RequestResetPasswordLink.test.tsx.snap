// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RequestResetPasswordLink > should match the snapshot 1`] = `
<div>
  <header
    class="css-cyjglw"
  >
    <div
      class="chakra-container css-old1by"
    >
      <div
        class="chakra-stack css-2jbh02"
      >
        <a
          href="/"
        >
          <p
            class="chakra-text css-kkaoep"
          >
            Trivia
          </p>
        </a>
        <div
          class="css-j0ozid"
        >
          <a
            href="/login"
          >
             Sign in 
          </a>
          <a
            href="/sign-up"
          >
             Sign Up 
          </a>
        </div>
      </div>
    </div>
  </header>
  <main
    class="chakra-container css-3i1d8n"
  >
    <div
      class="chakra-card css-132w0zy"
    >
      <div
        class="css-1updlva"
      >
        <p
          class="chakra-text css-mvfx24"
        >
          Password recovery
        </p>
        <form
          class="chakra-stack css-2umoaz"
          role="form"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="email"
              id="field-:r0:-label"
            >
              Email address
            </label>
            <input
              class="chakra-input css-0"
              id="email"
              name="email"
              placeholder="Enter your email"
              type="email"
            />
          </div>
          <button
            class="chakra-button css-uxt1e8"
            type="submit"
          >
            Send
          </button>
        </form>
      </div>
      <img
        alt="Login image"
        class="chakra-image css-1b9ujuc"
        src="/src/assets/recovery.svg"
      />
    </div>
  </main>
  <footer
    class="css-1s2lyti"
  >
    <div
      class="chakra-container css-0"
    >
      <div
        class="chakra-stack css-11h4xmw"
      >
        <a
          href="/"
        >
          <p
            class="chakra-text css-rekzoj"
          >
            Trivia
          </p>
        </a>
        <div
          class="css-0"
        >
          <strong
            class="chakra-heading css-kf3i1p"
          >
            Site map
          </strong>
          <div
            class="chakra-stack css-1ko7mmx"
          >
            <a
              href="/"
            >
              Home
            </a>
            <a
              href="/login"
            >
              Sign in
            </a>
            <a
              href="/sign-up"
            >
              Sign up
            </a>
          </div>
        </div>
        <div
          class="css-0"
        >
          <strong
            class="chakra-heading css-kf3i1p"
          >
            Social
          </strong>
          <div
            class="chakra-stack css-1ko7mmx"
          >
            <a
              href="https://www.linkedin.com/in/ingrid-flack/"
            >
              LinkedIn
            </a>
            <a
              href="https://github.com/ingridflack"
            >
              GitHub
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1y3yl0q"
      >
        <p
          class="chakra-text css-0"
        >
          Â© 2024 Trivia App
        </p>
        <p
          class="chakra-text css-0"
        >
          Developed by Ingrid Flack
        </p>
      </div>
    </div>
  </footer>
</div>
`;
